src/Redux/Dragons/Dragons.js

```diff
@@ -2,13 +2,19 @@ import { createAsyncThunk } from '@reduxjs/toolkit';
 import axios from 'axios';
 
 const GET_DRAGON = 'GET_DRAGON';
+const BOOK_DRAGON = 'BOOK_DRAGON';
 const initialState = [];
 const baseURL = 'https://api.spacexdata.com/v3/dragons';
 
 const reducerDragons = (state = initialState, action) => {
   switch (action.type) {
     case `${GET_DRAGON}/fulfilled`:
       return [...state, ...action.payload];
+    case `${BOOK_DRAGON}`:
+      return state.map((dragon) => {
+        if (dragon.id !== action.payload) return { ...dragon, reserved: true };
```

Comment:
- [ ] The if condition should be:

`if (dragon.id === action.payload)` instead of `if (dragon.id !== action.payload)`


**When I removed the `!`, your code worked correctly.**